client-runtime/src/main/java/com/microsoft/rest/v2/policy/RequestPolicyChain.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.rest.v2.policy;
+
+import com.microsoft.rest.v2.http.HttpClient;
+import com.microsoft.rest.v2.http.HttpRequest;
+import com.microsoft.rest.v2.http.HttpResponse;
+import rx.Single;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Type representing request-response pipeline holding request policies.
+ */
+public class RequestPolicyChain extends HttpClient {
+    private final List<RequestPolicy.Factory> factories;
+
+    /**
+     * Creates RequestPolicyChain.
+     *
+     * @param factoryArray the factories that can creates RequestPolicy instances in the pipeline.
+     */
+    public RequestPolicyChain(RequestPolicy.Factory... factoryArray) {
+        factories = Arrays.asList(factoryArray);
+        Collections.reverse(factories);
+    }
+
+    /**
+     * Creates RequestPolicyChain.
+     *
+     * @param factories the factories that can creates RequestPolicy instances in the pipeline.
+     */
+    public RequestPolicyChain(List<RequestPolicy.Factory> factories) {
+        this.factories = new ArrayList<>(factories);
+        Collections.reverse(this.factories);
```

Comment:
In fact, rather than reverse the collection here, why not reverse it in create()? That way you only call Collections.reverse() once, and it puts the reasoning for reversing the collection next to where the reversed collection will be used.

Reply:
.create() is expected to be called per-request.